src/analysis/functions.rs

```diff
@@ -206,14 +210,32 @@ fn analyze_function(
     configured_functions: &[&config::functions::Function],
     imports: &mut Imports,
 ) -> Info {
-    let async = func.parameters.iter().any(|parameter| parameter.scope == ParameterScope::Async);
+    let async = func.name.ends_with("_async") ||
+                func.parameters.iter()
+                               .any(|parameter| parameter.scope == ParameterScope::Async &&
+                                                parameter.c_type == "GAsyncReadyCallback");
+    let expecting_data = !async &&
+                         func.parameters.iter()
+                                        .any(|par| par.c_type.ends_with("Func") ||
+                                                   par.c_type.ends_with("Callback"));
```

Comment:
Same as above, don't guess based on partial type names but use the indexes from the gir